envoy/network/connection_handler.h

```diff
@@ -17,6 +17,28 @@
 namespace Envoy {
 namespace Network {
 
+// An effectively abstract class, used for dynamic class inheritance via dynamic_cast
+// such that options specific to individual listener types can be passed in
+// through the shutdownListener function, without other listener types needing
+// to know about it. Subclasses should specify `virtual ExtraShutdownListenerOptions`
+// so that they can be combined with other subclasses into a class containing multiple
+// types of options.
+class ExtraShutdownListenerOptions {
+public:
+  virtual ~ExtraShutdownListenerOptions() = default;
+};
+
+// An interface for ExtraShutdownListenerOptions enabling us to forward UDP packets from
+// the parent instance to the child instance during hot restart draining.
+// This is currently supported for QUIC listeners.
+// TODO(mattklein123): determine if other UDP ports have a reason to do this.
+class HotRestartUdpPacketForwarding : virtual public ExtraShutdownListenerOptions {
+public:
+  virtual void forwardUdpPacket(uint32_t worker_index, const Network::UdpRecvData& packet) PURE;
+};
+
+using ExtraShutdownListenerOptionsPtr = std::shared_ptr<ExtraShutdownListenerOptions>;
```

Comment:
nits: ExtraShutdownListenerOptionsSharedPtr. *Ptr is usually used in alias for unique_ptr.

Reply:
Done.